app/controllers/votes_controller.rb

```diff
@@ -32,31 +32,22 @@ def new
 
   def create
     if(params[:upvote]) 
-      p "user id class: #{current_user.id.class}  place id class: #{params[:place_id].class}  photourl class: #{params[:photourl].class}"
-      Upvote.create(user_id: current_user.id, place_id: params[:place_id], photourl: params[:photourl])
-      p "you created a new upvote"
-      redirect_to vote_path
+        Upvote.create(user_id: current_user.id, place_id: params[:place_id], photourl: params[:photourl])
+        p "you created a new upvote"
+        redirect_to vote_path
     elsif(params[:downvote])
-      Downvote.create(user_id: current_user.id, place_id: params[:place], photourl: params[:photourl])
-      p "you created a new downvote"
-      redirect_to vote_path
+        Downvote.create(user_id: current_user.id, place_id: params[:place], photourl: params[:photourl])
+        p "you created a new downvote"
+        redirect_to vote_path
     else
+      flash[:notice] = "Unable to retrieve photo" 
       redirect_to vote_path
```

Comment:
I see `redirect_to vote_path` 3 times here.  Maybe it could be moved to after the `if...end`